src/components/NavBar/NavigationBar.tsx

```diff
@@ -131,54 +150,52 @@ const ResponsiveAppBar: React.FC<Props> = ({ children }) => {
         setDrawer(false);
     };
 
-    const pages = [
-        ["Clients", "/clients"],
-        ["Lists", "/lists"],
-        ["Calendar", "/calendar"],
-        ["Admin", "/admin"], // TODO VFB-23 Prevent non-admins from viewing and accessing this page
-    ];
+    const { role } = useContext(RoleUpdateContext);
 
     return (
         <>
             <LoginDependent>
                 <StyledSwipeableDrawer open={drawer} onClose={closeDrawer} onOpen={openDrawer}>
                     <DrawerInner>
                         {pages.map(([page, link]) => (
-                            <DrawerButtonWrapper key={page}>
-                                <UnstyledLink href={link} onClick={closeDrawer} prefetch={false}>
-                                    <DrawerButton variant="text">{page}</DrawerButton>
-                                </UnstyledLink>
-                            </DrawerButtonWrapper>
+                            <RoleDependent role={role} key={page} pathname={link}>
+                                <DrawerButtonWrapper>
+                                    <UnstyledLink
+                                        href={link}
+                                        onClick={closeDrawer}
+                                        prefetch={false}
+                                    >
+                                        <DrawerButton variant="text">{page}</DrawerButton>
+                                    </UnstyledLink>
+                                </DrawerButtonWrapper>
+                            </RoleDependent>
                         ))}
                     </DrawerInner>
                 </StyledSwipeableDrawer>
             </LoginDependent>
             <AppBar>
                 <AppBarInner>
                     <MobileNavMenuContainer>
-                        <LoginDependent>
```

Comment:
do we not need a `LoginDependent` any more? what about a `RoleDependent`? Should mobile users only see the buttons they are meant to click?